homeassistant/components/device_tracker/bluetooth_tracker.py

```diff
@@ -17,24 +17,31 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-REQUIREMENTS = ['pybluez==0.22']
+REQUIREMENTS = ['pybluez==0.22', 'bt_proximity==0.1.2']
 
 BT_PREFIX = 'BT_'
 
+CONF_REQUEST_RSSI = 'request_rssi'
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_TRACK_NEW): cv.boolean
+    vol.Optional(CONF_TRACK_NEW): cv.boolean,
+    vol.Optional(CONF_REQUEST_RSSI): cv.boolean
 })
 
 
 def setup_scanner(hass, config, see, discovery_info=None):
     """Set up the Bluetooth Scanner."""
     # pylint: disable=import-error
     import bluetooth
+    from bt_proximity import BluetoothRSSI
 
     def see_device(device):
         """Mark a device as seen."""
+        attributes = {}
+        if len(device) > 2:
+            attributes['rssi'] = device[2]
```

Comment:
Btw, in general this is a horrible way to make an API; it is ambiguous and if the order of parameters changes at some point everything goes down with a fire. For now it's okay, but for the future you may want to look into attrs library (http://www.attrs.org/en/stable/index.html) or namedtuples instead of plain old tuples or lists to deliver such info.

Reply:
Most of the tuple stuff here isn't that pretty, it was simply done in an effort to keep as much code from the previous version as possible. Maybe I should simply add the three variables as parameters to the `see_device` method? That is probably the most readable solution?

Reply:
Like this:

```python
def see_device(mac, name, rssi):
        """Mark a device as seen."""
        attributes = {}
        if rssi is not None:
            attributes['rssi'] = rssi
        see(mac=BT_PREFIX + mac, host_name=name,
            attributes=attributes, source_type=SOURCE_TYPE_BLUETOOTH)
```

Also cleans up tuple stuff in the `update_bluetooth` method.